src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -858,6 +858,11 @@ public override int GetHashCode()
 
         private static bool EqualsCore(in Guid left, in Guid right)
         {
+            if (Vector128.IsHardwareAccelerated)
+            {
+                return Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(in left)) == Unsafe.As<Guid, Vector128<byte>>(ref Unsafe.AsRef(in right));
```

Comment:
```suggestion
                return Unsafe.ReadUnaligned<Vector<128>>(ref Unsafe.As<Guid, byte>(ref Unsafe.AsRef(in left))) == Unsafe.ReadUnaligned<Vector<128>>(ref Unsafe.As<Guid, byte>(ref Unsafe.AsRef(in right)));
```
This should use ReadUnaligned for portability.

Reply:
I suspect it can use the new `Vector128.LoadUnsafe` API as well

Reply:
Yes - I have updated the suggestions.

Reply:
Is there somewhere I can read about why unaligned reads are necessary here but not for the `Unsafe` operations below?

Reply:
Guid has int field. It is safe to assume that it is sufficiently aligned for int.